src/main/java/com/atlassian/db/replica/spi/ReplicaConsistency.java

```diff
@@ -12,12 +12,24 @@
 public interface ReplicaConsistency {
 
     /**
-     * Informs that {@code main} received an UPDATE, INSERT or DELETE.
+     * Informs that {@code main} received an UPDATE, INSERT or DELETE or transaction commit
+     * when in a transaction.
      *
      * @param main connects to the main database
      */
     void write(Connection main);
 
+    /**
+     * Invoked just before transaction commit.
+     *
+     * Notice: The method will not handle all writes. Writes done outside of a transaction
+     * needs to be handled in `ReplicaConnection#write`.
+     *
+     * @param main connects to the main database
+     */
+    default void preWrite(Connection main){
+    }
+
```

Comment:
It there is a need for `preWrite` maybe there is a need for `afterWrite`
How about an interceptor?

Reply:
`afterWrite` = write. I thought about renaming it. We would have to deprecate `write`. I'm still not convinced if it's worth it. @dagguh WDYT?

Reply:
Clarity is worth a deprecation

Reply:
I guess it should be
`write` and `preCommit`. 
`afterWrite` and `preWrite` would be misleading. It would seem to be symmetrical, while it's not. We can hook before commit only if there's a transaction. `preWrite`/`preCommit` is a subset of `write`.